### DISCUSSION TITLE
Remove camelCase enforcement

### DISCUSSION THREAD

**Discussion by 0x241F31** [capnproto/capnproto#2128] [Ideas]
I think it would be nice to remove hard-coded camelCase enforcement and make it configurable in generators.
For example, protobuf recommends snake_case for fields but it actually allows any other formats.

Actually, I just removed this branch https://github.com/capnproto/capnproto/blob/e02b77b9275f3f74fe71af1e2fcd7e32320e18f0/c%2B%2B/src/capnp/compiler/node-translator.c%2B%2B#L787

and seems like it works at first glance.

**Comment by dwrensha** [comment]
Note that the C++ and Rust code generators have `name` annotations that allow generated field names to be set: https://github.com/capnproto/capnproto/blob/e02b77b9275f3f74fe71af1e2fcd7e32320e18f0/c%2B%2B/src/capnp/c%2B%2B.capnp#L26
https://github.com/capnproto/capnproto-rust/blob/e068cba1938c945aef151f8422cea03c5009e082/capnpc/rust.capnp#L9

**Comment by kentonv** [comment]
Code generators for specific languages should convert camel-case names into the appropriate naming style for the target language.

But, Cap'n Proto schemas themselves must always use camel-case. This rule is enforced in order to ensure that code generators do not need to worry about supporting multiple styles as input.
